src/types/responses/ValidateAddress.ts

```diff
@@ -1,4 +1,5 @@
 export interface IValidateAddrResponse {
 	isValid: boolean;
 	ss58Prefix: number | null;
+	networkName: string | null;
```

Comment:
I would also add `AccountId` that would return the account stripped of any formatting. In the case of the Relay Chain, it's just a 32 byte value.

Perhaps it is a separate endpoint, but there could be a prefix converter that would take args like `?public=true` for giving an `AccountId` as input and `?prefix=<int>` for defining the prefix.

Reply:
@joepetrowski @TarikGul So in practice that would be something like this ?
- If the given address is a substrate address like 
  ```
  5Gv8YYFu8H1btvmrJy9FjjAWfb99wrhV3uhPFoNEr918utyR
  ``` 
  then the request would look like this : 
  ```
  http://127.0.0.1:8080/accounts/5Gv8YYFu8H1btvmrJy9FjjAWfb99wrhV3uhPFoNEr918utyR/validate
  ```
  and the response will look like :  
  ```
  isValid : true
  ss58Prefix : "42"
  networkName : "substrate"
  accountId : "0x2ad6a3105d6768e956e9e5d41050ac29843f98561410d3a47f9dd5b3b227ab87"
  ```
  and that would mean that args `public=false` 

- If what is given is a substrate `accountId` like 
  ```
  0x2ad6a3105d6768e956e9e5d41050ac29843f98561410d3a47f9dd5b3b227ab87
  ``` 
  then the request would look like : 
  ```
  http://127.0.0.1:8080/accounts/0x2ad6a3105d6768e956e9e5d41050ac29843f98561410d3a47f9dd5b3b227ab87?public=true&prefix=42/validate
  ```
  and the response will look like :  
  ```
  isValid : true
  ss58Prefix : "42"
  networkName : "substrate"
  address : "5Gv8YYFu8H1btvmrJy9FjjAWfb99wrhV3uhPFoNEr918utyR"
  ```
 
My apologies if i misunderstood the requirement and thank you so much for any feedback! :))

Reply:
Yes, exactly! I'd also add in conversion, so:

```
http://127.0.0.1:8080/accounts/5Gv8YYFu8H1btvmrJy9FjjAWfb99wrhV3uhPFoNEr918utyR/validate?prefix=0
```

Would return:

```
isValid : true
ss58Prefix : "0"
networkName : "polkadot"
address : "15rRgsWxz4H5LTnNGcCFsszfXD8oeAFd8QRsR6MbQE2f6XFF"
accountId : "0xd6a3105d6768e956e9e5d41050ac29843f98561410d3a47f9dd5b3b227ab8746"
```

Also starting to realize that this might be getting more complicated than expected. To do this, you also need a `?scheme` param that takes `{sr25519, ed25519, ecdsa}`, and `if ecdsa && public` then the account ID is the hash of the public key.

Reply:
@joepetrowski I think this should be a separate endpoint so we don't couple the logic between validation and conversion. My thought would be:

`/accounts/{accountId}/convert`

It would take in 3 query params: `scheme` = sr25519, `prefix` = 42, `public` = true. Each value to the right of the equal sign would be the defaults. This way it's clear to users what the REST endpoint is providing, and would also be easier to maintain.

Reply:
Yeah that's reasonable.